ds_judgements_public_ui/templates/includes/results_applied_filters.html

```diff
@@ -0,0 +1,38 @@
+{# {% if form.search_term.data or (form.collections.data != 'all') or (form.courts.data != 'all') %} #}
+  <h2 class="results-search-component__sub-header">Filters include</h2>
+  <ul class="results-search-component__removable-options">
+    {# {% if form.search_term.data %} #}
+      <li>
+        <span class="results-search-component__removable-options-key">Search term:</span>
+        <span class="results-search-component__removable-options-value">
+          <span class="results-search-component__removable-options-value-text">{# {{ form.search_term.data }} #}</span>
+          <a class="results-search-component__removable-options-link" href="{# {{ url_for('results', search_term='', collections=form.collections.data, courts=form.courts.data) }} #}">
```

Comment:
I've got a little template tag I use for this kind of thing:

```
from django import template
from django.db.models import Model
from django.http.request import QueryDict

register = template.Library()

MODE_ADD = "__add"
MODE_REMOVE = "__remove"
MODE_TOGGLE = "__toggle"


@register.simple_tag(takes_context=True)
def querystring_modify(
    context, base=None, remove_blanks=False, remove_utm=True, **kwargs
):
    """
    Renders a URL and IRI encoded querystring (e.g. "q=Hello%20World&amp;category=1") that is safe to include in links.
    The querystring for the current request (``request.GET``) is used as a base by default, or an alternative
    ``QueryDict``, ``dict`` or querystring value can be provided as the first argument. The base value can be modified
    by providing any number of additional key/value pairs. ``None`` values are discounted automatically, and blank
    values can be optionally discounted by specifying ``remove_blanks=True``.

    When specifying key/value pairs, any keys that do not already exist in the base value will be added, and those
    that do will have their value replaced. Specifying a value of ``None`` for an existing item will result in it being
    discounted. For example, if the querystring for the current request were "foo=ORIGINALFOOVAL&bar=ORIGINALBARVAL",
    but you wanted to:

    * Change the value of "foo" to "NEWFOOVAL"
    * Remove "bar"
    * Add a new "baz" item with the value `1`

    You could do so using the following:

    ```
    {% load querystring_modify %}
    {% querystring_modify foo="NEWFOOVAL" bar=None baz=1 %}```
    ```

    The output of the above would be "?foo=NEWFOOVAL&amp;baz=1".

    Values can be strings, booleans, integers or references to other variables in the current context.
    For example, if the tag were being used to generate pagination links, where the page number
    was a variable named ``page_num``, you could reference that value like so:

    ```
    {% load querystring_modify %}
    {% querystring_modify p=page_num %}
    ```

    You can also specify more than one value for a key by providing an iterable as a value. For example, if the context
    contained a variable ``tag_list``, which was list of 'tag' values (```['tag1', 'tag2', 'tag3']```), you include all
    of those values by referencing the list value. For example:

    ```
    {% load querystring_modify %}
    {% querystring_modify tags=tag_list %}
    ```

    The output of the above would be "?tags=tag1&amp;tags=tag2&amp;tags=tag3" (plus whatever other values were in the
    base value).

    And finally, if there are potentially multiple values for a specific parameter,
    and you only want to add or remove a single one, you can add '__add' or '__remove'
    to the variable name in your list of modifier kwargs.

    For example, if the querystring was "tags=tag1&amp;tags=tag2&amp;tags=tag3", and you wanted to remove 'tag2', you
    could do:

    ```
    {% load querystring_modify %}
    {% querystring_modify tags__remove='tag2' %}
    ```

    Which would output: "?tags=tag1&amp;tags=tag3"

    Or, if you wanted to add a new 'tagNew' value to that same parameter, you could do:

    ```
    {% load querystring_modify %}
    {% querystring_modify tags__add='tagNew' %}
    ```

    Which would output: "?tags=tag1&amp;tags=tag2&amp;tags=tag3&amp;tags=tagNew"

    Similarly, if you're unsure whether the value is present or not, and would like to remove it if it is,
    or add it if it is not, you can also use the '__toggle' option. For example:

    ```
    {% load querystring_modify %}
    {% querystring_modify tags__toggle='tagNew' %}
    ```

    You can add as many modifiers to the same tag as you need to, with any
    combination of modifiers at once. For example, the following is perfectly valid:

    ```
    {% load querystring_modify %}
    {% querystring_modify page=None tags__add="newTag" tags__remove="oldTag" %}
    ```

    Modifiers always fail gracefully if the value you're trying to add is already
    present, or a value you're trying to remove is not, or the named parameter isn't
    present at all.
    """
    querydict = get_base_querydict(context, base)
    for key, value in kwargs.items():

        if isinstance(value, Model):
            value = str(value.pk)
        elif not hasattr(value, "__iter__"):
            value = str(value)

        if key.endswith(MODE_TOGGLE):
            key = key[: -len(MODE_TOGGLE)]
            values = set(querydict.get_list(key))
            if value in values:
                values.remove(value)
            else:
                values.add(value)
            querydict.set_list(key, list(values))

        elif key.endswith(MODE_ADD):
            key = key[: -len(MODE_ADD)]
            values = set(querydict.get_list(key))
            if value not in values:
                values.add(value)
                querydict.set_list(key, list(values))

        elif key.endswith(MODE_REMOVE):
            key = key[: -len(MODE_REMOVE)]
            values = set(querydict.get_list(key))
            if value in values:
                values.remove(value)
                querydict.set_list(key, list(values))

        elif value is None:
            querydict.pop(key, None)
        else:
            if isinstance(value, (str, bytes)):
                querydict[key] = value
            elif hasattr(value, "__iter__"):
                querydict.setlist(key, list(value))

    clean_querydict(querydict, remove_blanks, remove_utm)

    return f"?{querydict.urlencode()}"


def get_base_querydict(context, base):
    if base is None and "request" in context:
        return context["request"].GET.copy()
    if isinstance(base, QueryDict):
        return base.copy()
    if isinstance(base, dict):
        return QueryDict.fromkeys(base, mutable=True)
    if isinstance(base, str):
        return QueryDict(base, mutable=True)
    # request not present or base value unsupported
    return QueryDict("", mutable=True)


def clean_querydict(querydict, remove_blanks=False, remove_utm=True):
    remove_vals = {None}
    if remove_blanks:
        remove_vals.add("")

    if remove_utm:
        for key in tuple(querydict.keys()):
            if key.lower().startswith("utm_"):
                querydict.pop(key)

    for key, values in querydict.lists():
        cleaned_values = [v for v in values if v not in remove_vals]
        if cleaned_values:
            querydict.setlist(key, cleaned_values)
        else:
            del querydict[key]
```

Reply:
This would ideally live in something like `core/templatetags/querystring_modify.py` (`core` being a new app for generic things like this, that don't really fit into a single app.) The `core` app would need a blank `__init__.py` to make it an 'app', and `core` would need to be added to the `INSTALLED_APPS` setting in order for the templatetags to be loaded.

Reply:
Sounds like a good refactoring for a followup PR sometime. Nice one.